src/Tests/Microsoft.NET.Build.Tests/SourceLinkTests.cs

```diff
@@ -0,0 +1,238 @@
+{
+    public class SourceLinkTests : SdkTest
+    {
+        private static readonly Guid s_embeddedSourceKindGuid = new("0E8A571B-6926-466E-B4AD-8AB04611F5FE");
+
+        public SourceLinkTests(ITestOutputHelper log)
+            : base(log)
+        {
+        }
+
+        private void CreateGitFiles(string repoDir, string originUrl, string commitSha = "1200000000000000000000000000000000000000")
+        {
+            var gitDir = Path.Combine(repoDir, ".git");
+            var headsDir = Path.Combine(gitDir, "refs", "heads");
+
+            Directory.CreateDirectory(gitDir);
+            File.WriteAllText(Path.Combine(gitDir, "HEAD"), "ref: refs/heads/master");
+            Directory.CreateDirectory(headsDir);
+
+            if (commitSha != null)
+            {
+                File.WriteAllText(Path.Combine(headsDir, "master"), commitSha);
+            }
+
+            if (originUrl != null)
+            {
+                File.WriteAllText(Path.Combine(gitDir, "config"), $"""
+                [remote "origin"]
+                    url = {originUrl}
+                """);
+            }
+
+            File.WriteAllText(Path.Combine(repoDir, ".gitignore"), """
+                [Bb]in/
+                [Oo]bj/
+                """);
+        }
+
+        private unsafe void ValidatePdb(string pdbPath, bool expectedEmbeddedSources)
+        {
+            // Validates that *.AssemblyAttributes.cs file is embedded in the PDB.
+
+            var pdb = File.ReadAllBytes(pdbPath);
+            fixed (byte* pdbPtr = pdb)
+            {
+                var mdReader = new MetadataReader(pdbPtr, pdb.Length);
+                var attrDocHandle = mdReader.Documents.Single(h => mdReader.GetString(mdReader.GetDocument(h).Name).EndsWith(".AssemblyAttributes.cs"));
+                var cdis = mdReader.GetCustomDebugInformation(attrDocHandle);
+
+                Assert.Equal(expectedEmbeddedSources, cdis.Any(h => mdReader.GetGuid(mdReader.GetCustomDebugInformation(h).Kind) == s_embeddedSourceKindGuid));
+            }
+        }
+
+        [Fact]
+        public void WithNoGitMetadata()
+        {
+            // We need to copy the test project to a directory outside of the SDK repo,
+            // otherwise we would find .git directory in the SDK repo root.
+
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("SourceLinkTestApp", testDestinationDirectory: Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()))
+                .WithSource();
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute().Should().Pass();
+
+            var intermediateDir = buildCommand.GetIntermediateDirectory();
+            intermediateDir.Should().NotHaveFile("SourceLinkTestApp.sourcelink.json");
+        }
+
+        /// <summary>
+        /// When creating a new repository locally we want the build to work and not report warnings even before the remote is set.
```

Comment:
This description (that the build shouldn't report warnings) seems to contradict the assertion (that it should include a warning).

Reply:
Ah, forgot to update the assert. Was testing it manually.